main.cpp

```diff
@@ -119,288 +154,357 @@ struct JumpStarter
         float consumePower( float currentDraw = 250 );
     };
 
-    //1)jumpstart a car 
     float jumpStartCar( DeviceToBeCharged car );
-    //2)power devices trough USB
     float chargePhone( DeviceToBeCharged phone );
-    //3)display the battery charge level
     int displayChargeLevel();
 
     DeviceToBeCharged car;
 };
-/*
-3)midi controller
-5 properties:
-    1)amount of keys
-    2)amount of knobs
-    3)display size
-    4)performance controls
-    5)weight 
-3 things it can do:
-    1)send midi data when keys pressed
-    2)display current patch
-    3)transpose octave
- */
+
+float JumpStarter::jumpStartCar( DeviceToBeCharged device )
+{
+    return currentChargeLevel - device.currentDraw*12/100;  //probably not correct formula
+}
+
+bool JumpStarter::DeviceToBeCharged::startEngine( int timesToAttempt )
+{
+    if( isItACar )
+    {
+        if( timesToAttempt <= 3 )   //will not allow more than 3 attempts
+        return true;
+        else return false;
+    }
+    else return false;    
+}
+
+float JumpStarter::DeviceToBeCharged::consumePower( float current )
+{
+    return current*12;    
+}
+
+int JumpStarter::displayChargeLevel()
+{
+    return currentChargeLevel*3/100;
+}
+///////////////////////////////////////////////////////////////////////
 struct MidiController
 {
-    //1)amount of keys
     int amountOfKeys = 49;
-    //2)amount of knobs
     int amountOfKnobs = 5;
-    //3)display size
     int displayCharacters = 10;
-    //4)performance controls
-    int performanceControls = 2;
-    //5)weight 
+    int performanceControls = 2; 
     float weight = 25;
 
-    //1)send midi data when keys pressed
     int sendMidi( int keyPressed );
-    //2)display current patch
     int displayPatch( int patchNumber );
-    //3)transpose octave
     int transposeOctave( int semitones );
-
 };
-/*
-4)air contitioning unit
-5 properties:
-    1)portable
-    2)cooling power in BTU
-    3)noise level
-    4)heather option
-    5)directional fan
-3 things it can do:
-    1)remove environment heat to control room temperature
-    2)heat a room
-    3)select the fan speed
- */
+
+int MidiController::sendMidi( int keyPressed )
+{
+    return keyPressed + 36;     // first C key is C1 36
+}
+
+int MidiController::displayPatch( int patchNumber)
+{
+    int bankNumber = patchNumber/8;
+    return bankNumber + ( patchNumber - bankNumber*8 );    
+}
+
+int MidiController::transposeOctave( int semitones )
+{
+    if ( semitones > 12 )
+    return semitones - 12;
```

Comment:
be consistent with your spacing of `( )`.  Look at the Readme.md coding standard.
- [ ] revise